src/main/java/com/google/sps/servlets/DataFromDatabase.java

```diff
@@ -0,0 +1,134 @@
+package com.google.sps.servlets;
+
+import com.google.cloud.ByteArray;
+import com.google.cloud.spanner.DatabaseClient;
+import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.spanner.Statement;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/data-from-db")
+public class DataFromDatabase extends HttpServlet {
+
+  DatabaseClient dbClient;
+  private String[] selectedTables;
+
+  public void init() {
+    Spanner spanner = SpannerOptions.newBuilder().build().getService();
+    DatabaseId db = DatabaseId.of("play-user-data-beetle", "test-instance", "example-db");
+    this.dbClient = spanner.getDatabaseClient(db);
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html;");
+
+    for (String table : selectedTables) {
+      String colQuery = "SELECT column_name, spanner_type, is_nullable FROM information_schema.columns WHERE table_name = '" + table + "'";
+
+      try (ResultSet resultSet =
+        dbClient.singleUse().executeQuery(Statement.of(colQuery))) {
+
+        List<String> colnames = new ArrayList<>();
+        List<String> spannerTypes = new ArrayList<>();
+        List<String> nullables = new ArrayList<>();
+
+        while (resultSet.next()) {
```

Comment:
Should we return an error if there are no columns? (e.g. table doesn't exist in database)

Reply:
I'll add this in the next PR if that's ok? I modified this part in a new branch to use the AutoValue object, and I think if I do this error checking there it'll reduce merge conflicts.

Reply:
Done - threw a runtime exception if no columns for now